R/beam.R

```diff
@@ -24,26 +25,235 @@
 #' latitude only a small correction. Using default values assumes an average
 #' earth's radius of 6371 km.
 beam_height <- function(range, elev, k = 4 / 3, lat = 35, re = 6378, rp = 6357) {
+  assert_that(is.numeric(range))
+  assert_that(is.numeric(elev))
+  assert_that(is.number(k))
+  assert_that(is.number(lat))
+  assert_that(is.number(re))
+  assert_that(is.number(rp))
   sqrt(range^2 + (k * earth_radius(re, rp, lat))^2 +
     2 * range * (k * earth_radius(re, rp, lat)) * sin(elev * pi / 180)) - k * earth_radius(re, rp, lat)
 }
 
 earth_radius <- function(a, b, lat) {
   lat <- lat * pi / 180
-  sqrt(((a^2 * cos(lat))^2 + (b^2 * sin(lat))^2) / ((a * cos(lat))^2 + (b * sin(lat))^2))
+  1000*sqrt(((a^2 * cos(lat))^2 + (b^2 * sin(lat))^2) / ((a * cos(lat))^2 + (b * sin(lat))^2))
 }
 
 #' Calculate radar beam width
 #'
 #' Calculates the width of a radar beam as a function of range and beam angle.
 #'
-#' @param range numeric. Range (distance from the radar antenna) in km.
+#' @param range numeric. Range (distance from the radar antenna) in m.
 #' @param beam_angle numeric. Beam opening angle in degrees, typically the
 #' the angle between the half-power (-3 dB) points of the main lobe
 #'
 #' @return numeric. Beam width in m.
 #'
 #' @export
 beam_width <- function(range, beam_angle = 1) {
-  range * 1000 * sin(beam_angle * pi / 180)
+  assert_that(is.numeric(range))
+  assert_that(is.number(beam_angle))
+  range * sin(beam_angle * pi / 180)
 }
+
+#' Gaussian beam profile as a function of height
+#'
+#' Normalized altitudinal pattern of radiated energy as a function of
+#' altitude at a given distance from the radar.
+#'
+#' @inheritParams beam_height
+#' @inheritParams beam_width
+#' @param antenna numeric. Height of the centre of the radar antenna in meters
+#' @param height numeric. Height in meter.
+#'
+#' @return numeric.
+#'
+#' @details Beam profile is calculated using \link{beam_height} and \link{beam_width}. \code{height} and
+#' \code{antenna} should be given in reference to the same reference plane (e.g. ground level or sea level)
+#'
+#' @keywords internal
+gaussian_beam_profile=function(height,range,elev,antenna=0,beam_angle=1, k=4/3, lat=35, re = 6378, rp = 6357){
```

Comment:
@adokter to check if you're using consistent assignment (`<-`) and spacing, you can run:

```r
library(styler)
style_pkg()
```

after your edits.

Reply:
I've now run styler

Reply:
thanks!